### PR TITLE
feat: expose expected directives for e2e test

### LINKED ISSUE #1006: e2e: Expose expected config and e2e test cases
## Summary

Currently all users of the e2e tests have to copy the expected coraza config. As a user of the e2e test suite I would like to just access the expected config and test cases directly.

### Basic example

Replace [this](https://github.com/corazawaf/coraza/blob/main/testing/e2e/e2e_test.go#L26) with a public const inside https://github.com/corazawaf/coraza/blob/main/http/e2e/e2e.go 
Additionally all the e2e test cases could be exposed to be able to use the go built-in testing framework instead of calling an external binary.

### Motivation

Reducing complexity and overhead for tests

### PR DESCRIPTION
**Pull request by fionera**
Closes #1006 

**Make sure that you've checked the boxes below before you submit PR:**

- [ ] My code includes positive and negative tests.
- [ ] I have an appropriate description with correct grammar.
- [ ] I have read [Contribution guidelines](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#contribution-guidelines), [maintainers note](https://github.com/corazawaf/coraza/blob/master/CONTRIBUTING.md#maintainers) and [Quality standard](https://github.com/github.com/corazawaf/coraza/v3sso/coraza-waf/blob/master/CONTRIBUTING.md#quality-standards).
- [ ] My code is properly linted and passes pre-commit tests.

Thanks for your contribution :heart:

### CODE REVIEW

File: http/e2e/cmd/httpe2e/main.go [hunk 1]
```diff
@@ -19,24 +19,7 @@ import (
...
-SecRule REQUEST_URI "@streq /admin" "id:101,phase:1,t:lowercase,log,deny"
-SecRule REQUEST_BODY "@rx maliciouspayload" "id:102,phase:2,t:lowercase,log,deny"
-SecRule RESPONSE_HEADERS:pass "@rx leak" "id:103,phase:3,t:lowercase,log,deny"
-SecRule RESPONSE_BODY "@contains responsebodycode" "id:104,phase:4,t:lowercase,log,deny"
-# Custom rules mimicking the following CRS rules: 941100, 942100, 913100
-SecRule ARGS_NAMES|ARGS "@detectXSS" "id:9411,phase:2,t:none,t:utf8toUnicode,t:urlDecodeUni,t:htmlEntityDecode,t:jsDecode,t:cssDecode,t:removeNulls,log,deny"
-SecRule ARGS_NAMES|ARGS "@detectSQLi" "id:9421,phase:2,t:none,t:utf8toUnicode,t:urlDecodeUni,t:removeNulls,multiMatch,log,deny"
-SecRule REQUEST_HEADERS:User-Agent "@pm grabber masscan" "id:9131,phase:1,t:none,log,deny"
-*/
+// Expected Coraza config see http/e2e.go
```
Comment (M4tteoP) [inline http/e2e/cmd/httpe2e/main.go:22]: I would be a bit more verbose considering that loading these directives is a very needed step to run e2e and we need to make they actually take a look at the file we are pointing them 
```suggestion
// A dedicated set of directives is expected to be loaded for e2e testing. Refer to the `Directives` const in http/e2e.go.
```

**Review by M4tteoP — APPROVED** [review]
LGTM, thanks!

File: http/e2e/cmd/httpe2e/main.go (hunk 1 above)
Comment (fionera) [reply http/e2e/cmd/httpe2e/main.go]: Done Thanks 😄

File: http/e2e/cmd/httpe2e/main.go [hunk 2]
```diff
@@ -19,24 +19,7 @@ import (
...
-SecRule REQUEST_URI "@streq /admin" "id:101,phase:1,t:lowercase,log,deny"
-SecRule REQUEST_BODY "@rx maliciouspayload" "id:102,phase:2,t:lowercase,log,deny"
-SecRule RESPONSE_HEADERS:pass "@rx leak" "id:103,phase:3,t:lowercase,log,deny"
-SecRule RESPONSE_BODY "@contains responsebodycode" "id:104,phase:4,t:lowercase,log,deny"
-# Custom rules mimicking the following CRS rules: 941100, 942100, 913100
-SecRule ARGS_NAMES|ARGS "@detectXSS" "id:9411,phase:2,t:none,t:utf8toUnicode,t:urlDecodeUni,t:htmlEntityDecode,t:jsDecode,t:cssDecode,t:removeNulls,log,deny"
-SecRule ARGS_NAMES|ARGS "@detectSQLi" "id:9421,phase:2,t:none,t:utf8toUnicode,t:urlDecodeUni,t:removeNulls,multiMatch,log,deny"
-SecRule REQUEST_HEADERS:User-Agent "@pm grabber masscan" "id:9131,phase:1,t:none,log,deny"
-*/
+// A dedicated set of directives is expected to be loaded for e2e testing. Refer to the `Directives` const in http/e2e.go.
```
Comment (M4tteoP) [inline http/e2e/cmd/httpe2e/main.go:22]:
```suggestion
// A dedicated set of directives is expected to be loaded for e2e testing. Refer to the `Directives` const in http/e2e/e2e.go.
.
```

File: http/e2e/cmd/httpe2e/main.go [hunk 3]
```diff
@@ -19,24 +19,8 @@ import (
...
-SecRule REQUEST_BODY "@rx maliciouspayload" "id:102,phase:2,t:lowercase,log,deny"
-SecRule RESPONSE_HEADERS:pass "@rx leak" "id:103,phase:3,t:lowercase,log,deny"
-SecRule RESPONSE_BODY "@contains responsebodycode" "id:104,phase:4,t:lowercase,log,deny"
-# Custom rules mimicking the following CRS rules: 941100, 942100, 913100
-SecRule ARGS_NAMES|ARGS "@detectXSS" "id:9411,phase:2,t:none,t:utf8toUnicode,t:urlDecodeUni,t:htmlEntityDecode,t:jsDecode,t:cssDecode,t:removeNulls,log,deny"
-SecRule ARGS_NAMES|ARGS "@detectSQLi" "id:9421,phase:2,t:none,t:utf8toUnicode,t:urlDecodeUni,t:removeNulls,multiMatch,log,deny"
-SecRule REQUEST_HEADERS:User-Agent "@pm grabber masscan" "id:9131,phase:1,t:none,log,deny"
-*/
+// A dedicated set of directives is expected to be loaded for e2e testing. Refer to the `Directives` const in http/e2e/e2e.go.
+.
```
Comment (M4tteoP) [inline http/e2e/cmd/httpe2e/main.go:23]:
```suggestion

```

**Pull request merged**